File: java/command/VoiceAssistant.java
package command;

import java.util.HashMap;
import java.util.Map;

public class VoiceAssistant {
    private final Map<String, Command> commands;

    public VoiceAssistant() {
        this.commands = new HashMap<>();
    }

    public void setCommand(String talkingCommand, Command command) {
        commands.put(talkingCommand, command);
    }

    public void say(String talkingCommand) {
        commands.get(talkingCommand).execute();
    }
}

File: dart_lang/lib/b_behavioral/j_command/voice_assistant.dart
import 'package:dart_lang/b_behavioral/j_command/command_interface.dart';

class VoiceAssistant {
  final Map<String, CommandInterface> commands = {};
  void setCommand(String voiceCommandName, CommandInterface command) {
    commands[voiceCommandName] = command;
  }

  void openVoiceCommand(String voiceCommandName) {
    commands[voiceCommandName]!.execute();
  }

  void say(String voiceCommandName) {
    print('Voice Assistant: $voiceCommandName');
    commands[voiceCommandName]!.execute();
  }
}
